check-submodules.ps1

```diff
@@ -0,0 +1,147 @@
+$answeredAll = $false   # has the user answered "yes to all" to init'ing submodules?
+$ProjectRoot = $PSScriptRoot
+
+# ask for confirmation and initialize a submodule if approved.
+function init_submodule($Path) {
+  $done = $false
+  while (-Not $done) {
+    Write-Warning "submodule $Path does not appear to be initialized."
+    Write-Host "Should I initialize it for you [Yes/no/all/quit]"
+    $answer = [Console]::ReadKey()
```

Comment:
What happens if you're running the build noninteractively and a submodule isn't initialized? Will this hang?

Reply:
Yes, this would hang.  Is anything currently building non-interactive without a complete clean repo?  I can check for interactivity but calling the script from build.sh/ps1 counts as non-interactive, so it'd become a manual script only.

Reply:
I'm not aware of any places, but if it does happen even unintentionally, hangs are a pretty bad CI/etc. experience.

Reply:
Would it make sense to add a timeout on the input and then perform a default action so it wouldn't ever hang?

Reply:
That could work. `q` would be my default choice.

Reply:
Since initializing a submodule is "safe" (won't ever lose work), I'm thinking "yes to all" makes sense for initialization.  I was thinking `no` for update and delete (in case you start the build and walk away) but I'm fine either way.

Reply:
Good point, `a` works for me.